internal/services/netapp/netapp_backup_policy_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package netapp_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2024-03-01/backuppolicy"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type NetAppBackupPolicyResource struct{}
+
+func (t NetAppBackupPolicyResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := backuppolicy.ParseBackupPolicyID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.NetApp.BackupPolicyClient.BackupPoliciesGet(ctx, *id)
+
+	if err != nil {
+		if resp.HttpResponse.StatusCode == http.StatusNotFound {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(true), nil
```

Comment:
This can be simplified 
```suggestion
	resp, err := clients.NetApp.BackupPolicyClient.BackupPoliciesGet(ctx, *id)
	if err != nil {
		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
	}

	return pointer.To(resp.Model != nil), nil
```